src/Features/CSharp/Portable/CodeRefactorings/ConvertLocalFunctionToMethod/CSharpConvertLocalFunctionToMethodCodeRefactoringProvider.cs

```diff
@@ -42,34 +43,54 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
             var cancellationToken = context.CancellationToken;
             var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
 
-            var identifier = await root.SyntaxTree.GetTouchingTokenAsync(context.Span.Start,
-                token => token.Parent.IsKind(SyntaxKind.LocalFunctionStatement), cancellationToken).ConfigureAwait(false);
-            if (identifier == default)
+            var localFunction = await TryGetSelectedLocalFunction(context.Span, root, cancellationToken).ConfigureAwait(false);
+            if (localFunction == default)
             {
                 return;
             }
 
-            if (context.Span.Length > 0 &&
-                context.Span != identifier.Span)
+            if (!localFunction.Parent.IsKind(SyntaxKind.Block, out BlockSyntax parentBlock))
             {
                 return;
             }
 
-            var localFunction = (LocalFunctionStatementSyntax)identifier.Parent;
-            if (localFunction.Identifier != identifier)
-            {
-                return;
-            }
+            context.RegisterRefactoring(new MyCodeAction(CSharpFeaturesResources.Convert_to_method,
+                c => UpdateDocumentAsync(root, document, parentBlock, localFunction, c)));
+        }
 
-            if (!localFunction.Parent.IsKind(SyntaxKind.Block, out BlockSyntax parentBlock))
+        private async Task<LocalFunctionStatementSyntax> TryGetSelectedLocalFunction(TextSpan selectedSpan, SyntaxNode root, CancellationToken cancellationToken)
+        {
+            if (selectedSpan.IsEmpty)
             {
-                return;
+                // This branch is required to handle "C LocalFunction[||](C c)" -> root.FindNode return ParameterList
+                var identifier = await root.SyntaxTree.GetTouchingTokenAsync(selectedSpan.Start,
+                    token => token.Parent.IsKind(SyntaxKind.LocalFunctionStatement), cancellationToken).ConfigureAwait(false);
+
+                return (identifier == default)
+                    ? null
+                    : (LocalFunctionStatementSyntax)identifier.Parent;
             }
+            else
+            {
+                var enclosingNode = root.FindNode(selectedSpan);
+                if (!(enclosingNode is LocalFunctionStatementSyntax))
+                {
+                    return null;
+                }
 
-            context.RegisterRefactoring(new MyCodeAction(CSharpFeaturesResources.Convert_to_method,
-                c => UpdateDocumentAsync(root, document, parentBlock, localFunction, c)));
+                var localFunction = (LocalFunctionStatementSyntax)enclosingNode;
+
+                if (selectedSpan != localFunction.Span && selectedSpan != localFunction.Identifier.Span)
```

Comment:
Why do we need this check? This will bail out if user selected all but the first char of return type OR selected whole method except the closing brace.

Reply:
I planned to handle that with abovementioned helper. The reason why we need some sort of check is that not having it would enable the diagnostics for situations such as: 

```
      void Whatever()
        {
        [| 
           // Some code
        |]
        }

```

which I thgouth is something we don't want.

Reply:
wouldn't FindNode there only find the block?

Reply:
The above selection will return `BlockSyntax` node to be returned.

Reply:
Nvm, there's a codeblock node in the way. Will fix.

Reply:
Ok, the change fails pre-existing test for `TestMissingAsync("C Local[|Function|](C c)");`. To me it makes sense to provide the refactoring in that case, however. 

Should I remove that test?

Reply:
> Should I remove that test?

I would suggest adjust the test to match the new expected behavior.

Reply:
Sure.

Reply:
Yeah, please never remove a test.  Just update accordingly :)